utils/validate_release_links.py

```diff
@@ -72,28 +74,47 @@ def get_release_tag() -> Optional[str]:
     raise RuntimeError("Can't determine release tag")
 
 
-def check_file(filename: str, global_allow_pattern: Pattern) -> List[str]:
+def check_file(
+    filename: str, global_allow_pattern: Pattern, release_tag: str
+) -> List[str]:
     """
     Validate a single file and return any offending lines.
     """
     bad_lines = []
-    with open(filename) as f:
-        for line in f:
-            if not RELEASE_PATTERN.search(line):
-                continue
-
-            if global_allow_pattern.search(line):
-                continue
-
-            if filename in ALLOW_LIST:
-                if ALLOW_LIST[filename] is None or ALLOW_LIST[filename].search(line):
-                    continue
+    new_file_name = f"{filename}.orig"
```

Comment:
I think we can simplify this a bit with a `tempfile` (don't need to clean it up after, and name should be unique). But don't worry about it for this PR.